src/common/attachments/attachmentsTable.component.tsx

```diff
@@ -0,0 +1,218 @@
+import ClearIcon from '@mui/icons-material/Clear';
+import {
+  Box,
+  Button,
+  TableCellBaseProps,
+  TableRow,
+} from '@mui/material';
+import {
+  MaterialReactTable,
+  useMaterialReactTable,
+  type MRT_ColumnDef,
+} from 'material-react-table';
+import { MRT_Localization_EN } from 'material-react-table/locales/en';
+import React from 'react';
+import { AttachmentPostMetadataResponse } from '../../api/api.types';
+import { useGetAttachments } from '../../api/attachments';
+import {
+  COLUMN_FILTER_FUNCTIONS,
+  COLUMN_FILTER_MODE_OPTIONS,
+  COLUMN_FILTER_VARIANTS,
+  TableBodyCellOverFlowTip,
+  TableCellOverFlowTipProps,
+  TableHeaderOverflowTip,
+  displayTableRowCountText,
+  formatDateTimeStrings,
+  getInitialColumnFilterFnState,
+} from '../../utils';
+import { usePreservedTableState } from '../preservedTableState.component';
+
+export interface AttachmentTableProps {
+  entityId?: string;
+}
+
+function AttachmentsTable(props: AttachmentTableProps) {
+  const { entityId } = props;
+  const { data: attachments, isLoading: attachmentIsLoading } = useGetAttachments(entityId);
+
+  const columns = React.useMemo<MRT_ColumnDef<AttachmentPostMetadataResponse>[]>(() => {
```

Comment:
```suggestion
  const columns = React.useMemo<MRT_ColumnDef<AttachmentPostMetadataResponse>[]>(() => {
```

This is not the correct type for the attachment list component. Look at backend models / schemas

Reply:
I had tried using `AttachmentPostMetadata` originally but ran into problems with it (I can't remember what now). The reason I used `AttachmentPostMetadataResponse` is it has the `id` property (https://github.com/ral-facilities/inventory-management-system/blob/develop/src/api/api.types.tsx), like `APIImage` does which is used by the image gallery component columns.

Reply:
You need to create a new type that matches the backend type